### PR TITLE
[SYCL] Rename sycl-fusion to sycl-jit 

### PR DESCRIPTION
**Pull request by jchlanda**
This patch renames the directory (`sycl-fusion` to `sycl-jit`) as well
as the involved library names (following the same principle: `fusion` to
`jit`).
To keep the user facing names consistent buildbot switches were updated.
And finally the codeowners file had to reflect the directory change.

### CODE REVIEW

**Comment by jchlanda** [comment]
This is a follow up from the https://github.com/intel/llvm/pull/14280 where the library/directory rename was discussed.

**Comment by jchlanda** [comment]
Code formatting is failing bad for this patch, but I'm not sure if it should be trusted, seems like it's in python files which have never been formatted in the first place, for instance [conifgure.py](https://github.com/intel/llvm/actions/runs/10092465540/job/27906159493?pr=14762#step:10:28).

File: sycl-jit/passes/CMakeLists.txt
Comment (sommerlukas) [inline sycl-jit/passes/CMakeLists.txt:2]: Should we maintain capitalization of "JIT" here, i.e., `SYCLKernelJIT`?

**Review by sommerlukas — COMMENTED** [review]
One minor nit about capitalization. 

There's also a few things missing that should be renamed, e.g., the CMake option `SYCL_ENABLE_KERNEL_FUSION` and related options. Potentially also macros should be renamed, while preserving feature test macros from the extension with the name specified by the extension.

File: sycl-jit/passes/CMakeLists.txt
Comment (jchlanda) [reply sycl-jit/passes/CMakeLists.txt]: Agreed, will apply the same logic to `SYCLKerneJitPasses`.

**Comment by jchlanda** [comment]
> `SYCL_ENABLE_KERNEL_FUSION`

We've discussed it offline a bit, going to rename it to `SYCL_ENABLE_EXTENSION_JIT`.

**Comment by dm-vodopyanov** [comment]
@jchlanda FYI, a commit with a new test which uses `// REQUIRES: fusion` was merged to intel/llvm an hour ago. As I don't see the whole current PR (https://github.com/intel/llvm/pull/14762) in UI, I assume it should be also updated:

The file is `sycl/test-e2e/NewOffloadDriver/diamond_shape.cpp`
PR: https://github.com/intel/llvm/pull/14526

**Comment by jchlanda** [comment]
@sommerlukas I believe, with the last two commits, we should have the basic renaming done.

**Comment by jchlanda** [comment]
> @jchlanda FYI, a commit with a new test which uses `// REQUIRES: fusion` was merged to intel/llvm an hour ago. As I don't see the whole current PR (#14762) in UI, I assume it should be also updated:
> 
> The file is `sycl/test-e2e/NewOffloadDriver/diamond_shape.cpp` PR: #14526

I think that should be fine, as we decided not to change the kernel fusion feature test macro, [and that's what `REQUIRES: fusion` relies upon](https://github.com/jchlanda/llvm/blob/jakub/fusion_rename/sycl/test-e2e/lit.cfg.py#L645), @sommerlukas can correct me if I'm wrong. 

Thank you very much for raising it though!

**Review by sommerlukas — APPROVED** [review]

**Comment by sarnex** [comment]
@jchlanda Currently seeing this build failure
```
/__w/llvm/llvm/src/sycl-fusion/passes/kernel-fusion/SYCLSpecConstMaterializer.cpp:288:13: error: expected unqualified-id
  288 |       llvm::setCurrentDebugType(DEBUG_TYPE);
      |             ^
/__w/llvm/llvm/src/llvm/include/llvm/Support/Debug.h:70:32: note: expanded from macro 'setCurrentDebugType'
   70 | #define setCurrentDebugType(X) do { (void)(X); } while (false)
```

Is it related to this PR?

**Comment by sarnex** [comment]
Wow, I can't read, what a mistake. My bad.

**Comment by aelovikov-intel** [comment]
> Code formatting is failing bad for this patch, but I'm not sure if it should be trusted, seems like it's in python files which have never been formatted in the first place, for instance [conifgure.py](https://github.com/intel/llvm/actions/runs/10092465540/job/27906159493?pr=14762#step:10:28).

I have formatted all our python files in https://github.com/intel/llvm/pull/12240 when we first enabled the check.

**Comment by sommerlukas** [comment]
> > Code formatting is failing bad for this patch, but I'm not sure if it should be trusted, seems like it's in python files which have never been formatted in the first place, for instance [conifgure.py](https://github.com/intel/llvm/actions/runs/10092465540/job/27906159493?pr=14762#step:10:28).
> 
> I have formatted all our python files in #12240 when we first enabled the check.

Correct me if I'm wrong, but #12240 does not touch `buildbot/configure.py`, which is the offending file in the format CI run.

**Comment by aelovikov-intel** [comment]
> > > Code formatting is failing bad for this patch, but I'm not sure if it should be trusted, seems like it's in python files which have never been formatted in the first place, for instance [conifgure.py](https://github.com/intel/llvm/actions/runs/10092465540/job/27906159493?pr=14762#step:10:28).
> > 
> > 
> > I have formatted all our python files in #12240 when we first enabled the check.
> 
> Correct me if I'm wrong, but #12240 does not touch `buildbot/configure.py`, which is the offending file with most changes in the format CI run.

That would mean that I just missed it, but the intention was to format everything and have the check clean. Please land a PR to reformat the file and then rebase this PR on top of it.

**Review by AlexeySachkov — APPROVED** [review]

**Comment by jchlanda** [comment]
@intel/dpcpp-devops-reviewers would you mind having a look, I think it's only [codeowners](https://github.com/intel/llvm/pull/14762/files#diff-3d36a1bf06148bc6ba1ce2ed3d19de32ea708d955fed212c0d27c536f0bd4da7) and [detect changes](https://github.com/intel/llvm/pull/14762/files#diff-eedfbc3fbc59e6ae8d8ddb3a832918304cfa8ec02b2d560afc65537aad4bc96a) that are relevant here.

**Review by aelovikov-intel — APPROVED** [review]

File: sycl/doc/GetStartedGuide.md
Comment (bader) [inline sycl/doc/GetStartedGuide.md:312]:
```suggestion
To disable support for these features, follow the instructions for the Linux
```

**Review by bader — APPROVED** [review]
LGTM, just one typo.

**Review by hdelan — APPROVED** [review]

**Review by gmlueck — APPROVED** [review]
Change to Getting Started Guide LGTM.

**Review by omarahmed1111 — APPROVED** [review]

**Pull request merged**
